scheduler/core/programprovider/gpp/gppprogramprovider.py

```diff
@@ -1040,10 +983,16 @@ def parse_observation(self,
                 # Process the target environment.
                 target_env = data[target_env_keys[0]]
                 # Use the explicit base if available, otherwise the first target in the asterism
-                try:
-                    target_info = target_env[GppProgramProvider._TargetKeys.BASE]
-                except KeyError:
-                    target_info = target_env[GppProgramProvider._TargetKeys.ASTERISM][0]
+
+                explicit_base = target_env.get(GppProgramProvider._TargetKeys.BASE)
+                if explicit_base is None:
+                    asterism = target_env.get(GppProgramProvider._TargetKeys.ASTERISM)
+                    if asterism:
+                        target_info = asterism[0]
+                    else:
+                        target_info = None
+                else:
+                    target_info = explicit_base
 
```

Comment:
similarly to the other cases, this might be clearer

`explicit_base = target_env.get(GppProgramProvider._TargetKeys.BASE)
asterism = target_env.get(GppProgramProvider._TargetKeys.ASTERISM)
target_info = explicit_base or (asterism[0] if asterism else None)`